### PR TITLE
marking `MAlonzo.*` as `autogen-modules`; restoring`metatheory` to `cabal.project`

### PR DESCRIPTION
**Pull request by jmchapman**
With `metatheory` listed as a package in `cabal.project`
```
$ cabal v2-install language-plutus-core
```
gives this error message:
```
...
cabal: Error: Could not find module: MAlonzo.Code.Main with any suffix:
["gc","chs","hsc","x","y","ly","cpphs","hs","lhs","hsig","lhsig"]. If the
module is autogenerated it should be added to 'autogen-modules'.
```

The modules `MAlonzo.*` are autogenerated. I tried moving them from `other-modules` to `autogen-modules` in `metatheory/plc-agda.cabal` as suggested by the error message. It seems to work for me.

### CODE REVIEW

**Comment by jmchapman** [comment]
ok, this is not going as well as I had hoped...

**Comment by jmchapman** [comment]
I am able to run tests and build  and run `plc-agda` using:

```
$ cd metatheory
$ agda --compile --ghc-dont-call-ghc Main.lagda
$ cd ..
$ cabal v2-build metatheory
$ cabal v2-test metatheory
$ cd metatheory
$ cabal v2-run plc-agda -- --help
```

`cabal v2-install metatheory` doesn't work anymore as it doesn't include the generated MAlonzo.* modules in the sdist:

```
$ cabal v2-sdist metatheory -l
metatheory/LICENSE
metatheory/Main.hs
metatheory/NOTICE
metatheory/Opts.hs
metatheory/README.md
metatheory/Raw.hs
metatheory/Scoped.hs
metatheory/TestDetailed.hs
metatheory/TestSimple.hs
metatheory/plc-agda.cabal
```

But, I guess this isn't crucial and I am not sure whether the Agda code or the generated Haskell should really be included in a hypothetical standalone package.

File: cabal.project [hunk 1]
```diff
@@ -17,6 +17,7 @@ packages: language-plutus-core
           plutus-contract
           deployment-server
           iots-export
+	  metatheory
```
Comment (michaelpj) [inline cabal.project]: weird indentation

**Comment by michaelpj** [comment]
Yes, the `sdist` is a source distribution, so shouldn't include autogenerated files. Those should be created by the setup. But as discussed we don't really want to spend lots of time writing a custom setup to do the Agda extraction... Also, `sdist` really doesn't matter for `plc-agda` as I doubt it's ever going on Hackage.

File: cabal.project (hunk 1 above)
Comment (jmchapman) [reply cabal.project]: there was a tab, i removed it.

**Comment by jmchapman** [comment]
`sdist` does matter for `v2-install` though, which is annoying but manageable.  I will make an issue for making a custom setup/to record that you can't `v2-install` metatheory and mark it as low priority.

**Pull request merged**